source/scpd/tests/VTableTest.d

```diff
@@ -0,0 +1,75 @@
+/*******************************************************************************
+
+    Contains checking if the virtual methods are in the same order
+    using vtable.
+
+    Copyright:
+        Copyright (c) 2020 BOS Platform Foundation Korea
+        All rights reserved.
+
+    License:
+        MIT License. See LICENSE for details.
+
+*******************************************************************************/
+
+module scpd.tests.VTableTest;
+
+version (Windows) {} else:
+
+/// In case the virtual methods are in the correct order.
+unittest
+{
+    // abstract class, this is in the correct order.
+    extern (C++) class TestA
+    {
+    public:
+        ~this()
+        {
+        }
+        abstract void vfunc1();
+        abstract void vfunc2();
+    }
+
+    extern (C++) class BonTestA : TestA
+    {
+    public:
+        override void vfunc1() {}
+        override void vfunc2() {}
+    }
+
+    auto n = new BonTestA();
+    // The point of the method is the same as the expected value.
+    assert(n.__vptr[2] is &BonTestA.vfunc1);
+    assert(n.__vptr[3] is &BonTestA.vfunc2);
+}
+
+/// In case the virtual methods are in the incorrect order.
+unittest
+{
+    // abstract class, this is in the incorrect order.
```

Comment:
What do you mean, incorrect order? The vtable will be based on the order of the methods as defined in the **base** class, not the inherited class. It doesn't matter what the order is in the derived class.

In your `BonTestB` derived class you can reorder the methods and your tests will still pass. But these are the sort of tests that belong in the compiler's test-suite, not Agora.

Reply:
Yes, I agree with you.
This is not to solve the issue.
But this will allow us to understand the issue correctly.
In conclusion, I don't think it's necessary. I'll remove it.

Reply:
> But this will allow us to understand the issue correctly.

We could write some documentation about it, if it's unclear. I think we have an issue for that! Here: #560